### PR TITLE
fix(rich-text): moving list items down

### PR DESCRIPTION
**Pull request by z0al**
Moves list items to the correct position when pressing `TAB`.

#### Before:

tab-list-previously

#### After:

tab-list

### CODE REVIEW

File: packages/rich-text/src/plugins/List/transforms/moveListItemUp.ts [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+    if (!liParent) {
+      let toListPath;
+      try {
+        toListPath = Path.next(listPath);
+      } catch (err) {
+        return;
+      }
+
+      const condA = hasListChild(editor, liNode);
+      const condB = !isLastChild(list, liPath);
```
Comment (cezarsmpio) [inline packages/rich-text/src/plugins/List/transforms/moveListItemUp.ts:45]: What does it mean? Do we have better names?

File: packages/rich-text/src/plugins/List/transforms/moveListItemUp.ts [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+    const liParent = getAbove<TElement>(editor, {
+      at: listPath,
+      match: { type: getPluginType(editor, ELEMENT_LI) },
+    });
+    if (!liParent) {
+      let toListPath;
+      try {
+        toListPath = Path.next(listPath);
+      } catch (err) {
+        return;
```
Comment (cezarsmpio) [inline packages/rich-text/src/plugins/List/transforms/moveListItemUp.ts:41]: Do we do any transformation before returning? I mean, if not, then I think exiting it is safe to do.

File: packages/rich-text/src/plugins/List/transforms/moveListItemUp.ts (hunk 1 above)
Comment (z0al) [reply packages/rich-text/src/plugins/List/transforms/moveListItemUp.ts:45]: This file was copied as is from Plate. I made changes on the `moveListItemDown`

File: packages/rich-text/src/plugins/List/transforms/moveListItemUp.ts (hunk 2 above)
Comment (z0al) [reply packages/rich-text/src/plugins/List/transforms/moveListItemUp.ts:41]: What do you mean by transformations here?

File: packages/rich-text/src/plugins/List/transforms/moveListItemUp.ts (hunk 1 above)
Comment (z0al) [reply packages/rich-text/src/plugins/List/transforms/moveListItemUp.ts]: I removed it completely and used the helper directly from Plate. I thought I'd need to modify it at first but not anymore.

File: packages/rich-text/src/plugins/List/transforms/moveListItemUp.ts (hunk 2 above)
Comment (z0al) [reply packages/rich-text/src/plugins/List/transforms/moveListItemUp.ts]: I got rid of this file

**Review by cezarsmpio — APPROVED** [review]

**Pull request merged**